src/libraries/System.Runtime.Extensions/tests/System/Environment.GetCommandLineArgs.cs

```diff
@@ -70,5 +70,30 @@ public static int CheckCommandLineArgs(string[] args)
 
             return RemoteExecutor.SuccessExitCode;
         }
+
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void GetCommandLineArgs_Fallback_Returns()
+        {
+            if (PlatformDetection.IsNotMonoRuntime
+                && PlatformDetection.IsNotNativeAot
+                && PlatformDetection.IsWindows)
+            {
+                // Currently fallback command line is only implemented on Windows coreclr
+                RemoteExecutor.Invoke(CheckCommandLineArgsFallback).Dispose();
+            }
+        }
+
+        public static int CheckCommandLineArgsFallback()
+        {
+            // Clear the command line args set for managed entry point
+            var field = typeof(Environment).GetField("s_commandLineArgs", BindingFlags.Static | BindingFlags.NonPublic);
+            Assert.NotNull(field);
+            field.SetValue(null, null);
+
+            string[] args = Environment.GetCommandLineArgs();
+            Assert.NotEmpty(args);
```

Comment:
Can we compare the return value with the original command line? The new value should be a super-set of the original command line.